fluent-package/yum/fluent-package.spec.in

```diff
@@ -324,27 +310,31 @@ if [ -f "%{_sysconfdir}/prelink.conf" ]; then
 fi
 if [ $1 -eq 2 ]; then
   # install missing plugins during upgrading package
-  if [ -f %{local_plugin_migration} ]; then
-    echo "Checking network connectivity..."
-    curl --fail --silent -O https://rubygems.org/specs.4.8.gz
-    if [ $? -eq 0 ]; then
-      echo "Install missing plugins automatically..."
-      /usr/sbin/fluent-gem list | grep fluent-plugin | cut -d' ' -f1 > %{local_base_plugins}
-      # NOTE: Do not install already bundled plugins
-      diff -u %{local_base_plugins} %{local_plugin_migration} | grep '^+fluent' | sed -e 's/\+//' > %{local_missing_plugins}
-      if [ -s %{local_missing_plugins} ]; then
-        cat %{local_missing_plugins} | xargs sudo /usr/sbin/fluent-gem install --no-document
+  /usr/sbin/fluent-gem list | cut -d' ' -f1 > %{local_current_gems}
+  if [ -f %{local_base_gems} ]; then
+    diff -u %{local_base_gems} %{local_current_gems} | grep '^-' | sed -e 's/^-//g' > %{local_missing_gems}
+    if [ -s %{local_missing_gems} ]; then
+      echo "Try to install the following gems:"
+      cat %{local_missing_gems}
+      if ! curl --fail --silent -O https://rubygems.org/specs.4.8.gz; then
+	echo "No network connectivity..."
       else
-         echo "No need to install missing plugins..."
+	echo "Install missing dependency gems..."
+	while read gem
+	do
+	  if [ "$gem" = "readline-ext" ]; then
+	    # readline-ext is unbundled since ruby 3.3
+	    continue
+	  fi
+          if ! /usr/sbin/fluent-gem install --no-document $gem; then
+	    echo "Can't install missing gem automatically, please install $gem manually."
+          fi
+	done < %{local_missing_gems}
       fi
-      rm -f %{local_base_plugins}
-      rm -f %{local_missing_plugins}
-    else
-      echo "Can't install missing plugins automatically, please install the following plugins manually."
-      cat %{local_plugin_migration}
     fi
-    rm -f %{local_plugin_migration}
+    rm -f %{local_missing_gems}
   fi
+  rm -f %{local_current_gems}
```

Comment:
`local_base_gems` file also should be remove like DEB package?

Reply:
Thanks, it should be.